src/main/scala/com/databricks/labs/overwatch/env/Workspace.scala

```diff
@@ -53,6 +53,24 @@ class Workspace(config: Config) extends SparkSessionWrapper {
 
   }
 
+  // The below function is supposed to fetch data about DatabricksSQLEndPoints
+  def getDatabricksSqlDF: DataFrame = {
+
+    val jobsEndpoint = "sql/endpoints"
+
+    try {
+      ApiCall(jobsEndpoint, config.apiEnv)
+        .executeGet()
+        .asDF
+        .withColumn("organization_id", lit(config.organizationId))
+    } catch {
+      case e: Throwable => {
+        logger.log(Level.ERROR, "ERROR: Failed to execute jobs/list API call.", e)
+        spark.sql("select ERROR")
```

Comment:
the latest PR made some changes to the pattern here, see if you can incorporate those.